tests/common/fixtures/advanced_reboot.py

```diff
@@ -0,0 +1,510 @@
+                self.testbed['topo']['properties']['configuration_properties']['common']['nhipv4'],
+                self.testbed['topo']['properties']['configuration_properties']['common']['nhipv6'],
+            ]
+        else:
+            self.rebootData['nexthop_ips'] = None
+
+    def __validateAndBuildSadList(self):
+        '''
+        Validate sad list (preboot/inboot lists) member data
+        '''
+        prebootList = [] if self.prebootList is None else self.prebootList
+        inbootList = [] if self.inbootList is None else self.inbootList
+        sadList = [item for item in itertools.chain(prebootList, inbootList)]
+
+        for item in sadList:
+            if ':' not in item:
+                continue
+            itemCnt = int(item.split(':')[-1])
+            if 'bgp_down' in item:
+                assert itemCnt <= self.hostMaxLen, (
+                    'Bgp neigh down count is greater than or equal to number of VM hosts '
+                    'Current val = {0} Max val = {1}'
+                ).format(itemCnt, self.hostMaxLen)
+            if 'lag_down' in item:
+                assert itemCnt <= self.hostMaxLen, (
+                    'Lag count is greater than or equal to number of VM hosts. '
+                    'Current val = {0} Max val = {1}'
+                ).format(itemCnt, self.hostMaxLen)
+            if 'lag_member_down' in item:
+                assert itemCnt <= self.lagMemberCnt, (
+                    'Lag member count is greater than available number of lag members. '
+                    'Current val = {0} Available cnt = {1}'
+                ).format(itemCnt, self.lagMemberCnt)
+            if 'vlan_port_down' in item:
+                assert itemCnt <= self.vlanMaxCnt, (
+                    'Vlan count is greater than or equal to number of Vlan interfaces. '
+                    'Current val = {0} Max val = {}'
+                ).format(itemCnt, self.vlanMaxCnt)
+            if 'routing' in item:
+                assert itemCnt <= self.hostMaxCnt, (
+                    'Number of prefixes is greater than allowed max. '
+                    'Current val = {0} Max val = {1}'
+                ).format(itemCnt, self.hostMaxCnt)
+
+        # Adding None item if the sadList is empty in order to run the test case once when sad list is empty
+        self.rebootData['sadList'] = sadList if len(sadList) > 0 else [None]
+
+    def __transferTestDataFiles(self, data, ansibleHost):
+        '''
+        Convert data into json format and transfers json file to ansible host (ptfhost/duthost)
+        @param data: map that includedata source and json file name
+        @param ansibleHost: Ansible host that is receiving this data
+        '''
+        for item in data:
+            data_source = item['source']
+            filename = '/tmp/' + item['name'] + '.json'
+            file = open(filename, 'w')
+            file.write(json.dumps(data_source))
+            file.close()
```

Comment:
This is a helpful trick to ensure the file gets closed even if an I/O error occurs:
```suggestion
            with open(filename, 'w') as file:
                file.write(json.dumps(data_source))
```